gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/Policy.java

```diff
@@ -0,0 +1,367 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.resourcemanager;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A Google Cloud IAM Policy object
+ */
+public class Policy implements Serializable {
+
+  private static final long serialVersionUID = 3493286111316914094L;
+  private final List<Binding> bindings;
+  private final Integer version;
+  private final String etag;
+
+  public static enum MemberType {
+    ALL_USERS("allUsers"),
+    ALL_AUTHENTICATED_USERS("allAuthenticatedUsers"),
+    USER("user:"),
+    SERVICE_ACCOUNT("serviceAccount:"),
+    GROUP("group:"),
+    DOMAIN("domain:");
+
+    private final String prefix;
+
+    MemberType(String prefix) {
+      this.prefix = prefix;
+    }
+
+    String prefix() {
+      return prefix;
+    }
+  }
+
+  public enum RoleType {
+    OWNER,
+    EDITOR,
+    VIEWER;
```

Comment:
Enum is nice as long as the list is bounded and is not likely to change.
However, from [here](https://cloud.google.com/iam/docs/overview) it looks like it may include service specific values (e.g. "Publisher").